tools/respecDocWriter.js

```diff
@@ -1,135 +1,173 @@
- * @param {object} [whenToHalt] Allowing execution to stop without writing.
- * @param {boolean} [whenToHalt.haltOnError] Do not write if a ReSpec processing has an error.
- * @param {boolean} [whenToHalt.haltOnWarn] Do not write if a ReSpec processing has a warning.
  * @param {object} [options]
  * @param {number} [options.timeout] Milliseconds before processing should timeout.
- * @param {boolean} [options.disableSandbox] See https://peter.sh/experiments/chromium-command-line-switches/#no-sandbox
- * @param {boolean} [options.debug] Show the Chromium window with devtools open for debugging.
+ * @param {(error: RsError) => void} [options.onError] What to do if a ReSpec processing has an error. Does nothing by default.
+ * @param {(warning: RsError) => void} [options.onWarning] What to do if a ReSpec processing has a warning. Does nothing by default.
  * @param {boolean} [options.verbose] Log processing status to stdout.
- * @param {(error: RsError) => void} [options.onError] What to do if a ReSpec processing has an error. Logs to stderr by default.
- * @param {(warning: RsError) => void} [options.onWarning] What to do if a ReSpec processing has a warning. Logs to stderr by default.
- * @return {Promise<string>} Resolves with HTML when done writing. Rejects on errors.
+ * @param {boolean} [options.disableSandbox] See https://peter.sh/experiments/chromium-command-line-switches/#no-sandbox
+ * @param {boolean} [options.devtools] Show the Chromium window with devtools open for debugging.
+ * @return {Promise<{ html: string, errors: RsError[], warnings: RsError[] }>}
+ * @throws {Error} If failed to process.
  */
-async function fetchAndWrite(
-  src,
-  out,
-  whenToHalt = {},
-  {
+async function convertToHTML(src, options = {}) {
+  const {
     timeout = 300000,
-    disableSandbox = false,
     verbose = false,
-    debug = false,
-    onError = error =>
-      console.error(
-        colors.error(`💥 ReSpec error: ${colors.debug(error.message)}`)
-      ),
-    onWarning = warning =>
-      console.warn(
-        colors.warn(`⚠️ ReSpec warning: ${colors.debug(warning.message)}`)
-      ),
-  } = {}
-) {
+    disableSandbox = false,
+    devtools = false,
+  } = options;
+  if (typeof options.onError !== "function") {
+    options.onError = () => {};
+  }
+  if (typeof options.onWarning !== "function") {
+    options.onWarning = () => {};
+  }
+
   const timer = createTimer(timeout);
 
-  const log =
-    verbose && out !== null
-      ? msg => console.log(`[Timeout: ${timer.remaining}ms] ${msg}`)
-      : () => {};
+  const log = verbose
+    ? msg => console.log(`[Timeout: ${timer.remaining}ms] ${msg}`)
+    : () => {};
+
+  /** @type {RsError[]} */
+  const errors = [];
+  /** @type {RsError[]} */
+  const warnings = [];
+  const onError = error => {
+    errors.push(error);
+    options.onError(error);
+  };
+  const onWarning = warning => {
+    warnings.push(warning);
+    options.onWarning(warning);
+  };
```

Comment:
I wonder if we should curry these two? 

```JS
function errorReporter(errors, handler) {
    return error => {
        errors.push(error);
        handler(error)
    }
}
```

Reply:
Maybe it's fine here. Can avoid duplication as just two nearby instances. Reduces indirection

Reply:
Yeah, agree, it's fine.